aws_lambda_powertools/utilities/batch/base.py

```diff
@@ -89,5 +90,6 @@ def failure_handler(self, record: Any, exception: Exception):
             "fail", exceptions args, original record
         """
         entry = ("fail", exception.args, record)
+        self.exceptions.append(exception)
```

Comment:
Can you add a `logger.debug` here? No need to log the record/data for security reasons.